File: Lab01-Horner/lab1.py
def horner(coffs,x):
    result = coffs[-1]
    for i in range (len(coffs) -2,-1, -1):
        result = result * x + coffs[i]
    return result

print("Progam do obliczania wielomianu metodą Hornera")
coffs_dane = input("Podaj współczynniki wielomianu w kolejności od najmniejszej potęgi do największej, odddzielając spacją:")
coffs = [float(c)for c in coffs_dane.split()]
x = float(input("Podaj x:"))

value = horner(coffs,x)
print(f"W({x}) = {value}")

File: Lab01-Horner/lab1.m
disp('Progam do obliczania wielomianu metod¹ Hornera');
coffs_dane = input('Podaj wspó³czynniki wielomianu w kolejnoœci od najmniejszej potêgi do najwiêkszej, odddzielaj¹c spacj¹:','s');
coffs = str2num(coffs_dane);
%#ok<ST2NM>
x=input('Podaj x:');

n=length(coffs);
value=coffs(end);
for i=n-1:-1:1
    value=value*x+coffs(i);
end

fprintf('W(%g)=%g\n',x,value);